src/redux/reducers/student_view.js

```diff
@@ -0,0 +1,15 @@
+export default function change_tutors(state = {}, action) {
+  let new_state = Object.assign({}, state);
+
+  switch (action.type) {
+    case 'CHANGE_ONLINE_TUTORS':
+      new_state.filteredTutors = action.payload;
+      break;
+
+    case 'ON_SEARCH':
+      new_state = Object.assign(new_state, action.payload);
+      break;
+  }
+
```

Comment:
This works... but another way to do it would be just to return from inside the switch case instead of waiting till the end. Returning in the switch case is normally what I've seen online